Source/Android/jni/InfinityConfig.cpp

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include <jni.h>
+
+#include <array>
+#include <cstdint>
+
+#include "AndroidCommon/AndroidCommon.h"
+#include "Core/IOS/USB/Emulated/Infinity.h"
+#include "Core/System.h"
+
+extern "C" {
+
+JNIEXPORT jobject JNICALL
+Java_org_dolphinemu_dolphinemu_features_infinitybase_InfinityConfig_getFigureMap(JNIEnv* env,
+                                                                                 jclass clazz)
+{
+  auto& system = Core::System::GetInstance();
+
+  jclass hash_map_class = env->FindClass("java/util/HashMap");
```

Comment:
You should probably put all these lookups in IDCache, so they don't have to be duplicated between getFigureMap and getReverseFigureMap and between Skylanders and Infinity.